### PR TITLE
Feat: setup theme

### PR DESCRIPTION
**Pull request by efJerryYang**
See issue #17 .

### CODE REVIEW

**Comment by efJerryYang** [comment]
Hey @PraxTube  I've refactored the code to avoid circular imports, you can see the comment in #17 and the commit messages for more details. Now, you can try modifying the `print.py` (which consists the use of `xxx_output()`) by using `file.py`. I also invited you as a collaborator, please check the invitation link you received. Let me know if you run into any problems.

**Comment by PraxTube** [comment]
That looks good. I was able to import the `load_config` without circular imports. I also like the break up of input/output.
:+1:

**Review by PraxTube — APPROVED** [review]

**Comment by efJerryYang** [comment]
Great, you can work on this branch on your local as I have granted you access right to this repository (if you have accepted the invitation as a collaborator). I will draft a new release tomorrow morning regardless of whether this thread is finished.

**Comment by PraxTube** [comment]
Ah yes, thanks for the invitation :partying_face: 

Do you want me to implement the theme changes in this PR then?

**Comment by efJerryYang** [comment]
It depends on the available time you have, as this feature does not have any impact on my daily usage of this tool. It is okay to leave this PR here until one of us has some free time to work on it.

**Comment by PraxTube** [comment]
In that case I will work on it now. It shouldn't be too much work.

**Comment by PraxTube** [comment]
You might have complains with how I implemented the check of the config file. I think something like that is useful, though I am not quite sure. The alternative would be for the user to remove the config file and recreate it, which would mean they either need to create a new API key or save the old one somewhere before pasting it into the new file.

Actually, not that I am thinking about it, it might not be that bad to just tell the user to remove the config file (or even do it automatically with the users approving it) and then to create a new config file.

**Comment by efJerryYang** [comment]
I will check your implementation later when I have free time (possibly not today). Thank you for your contributions.

**Comment by efJerryYang** [comment]
More changes are needed for this PR to be merged, see the commit messages:

```markdown
In our earlier implementation, we would update the dirty config by directly
creating a new one, but this may confuse the user. So we implemented a new
function to handle the dirty config update. You can see:
- `update_dirty_config` in `file.py`: update the dirty config
- `show_dirty_config_panel` in `panel.py`: prompt the user to update the
  dirty config

This commit also made modification to `check_config_data`, replacing the
`dirty_flag` with `missing_keys` list, so that can update missing ones
accordingly.

Notice: two TODOs are needed to be done in the future before merging this
PR, you can search `TODO` from the source code:
1. Add prompt to setup code theme, give a list of available themes
   and display a preview
2. Add more checks for specific keys, for example, `openai.api_key`
```

**Comment by efJerryYang** [comment]
Additionally, to make the warning information visible when the tool is launched, we might consider simplifying the welcome panel and separating it from the help panel.

**Comment by efJerryYang** [comment]
I force-pushed the rebased branch after the update of the main branch. Nothing changed actually.

**Comment by PraxTube** [comment]
I actually think it would be a good idea to split the welcome message and the help message. Maybe the help message can be displayed on you first launch of the program, but I is rather inconvenient that it shows you the whole help message whenever you open a new instance.

**Comment by PraxTube** [comment]
Maybe we can use the dict from `create_config_yaml()` in the checks of `check_config_data()`.

That way we also wouldn't need to hard-code the config twice.

**Comment by PraxTube** [comment]
Okay so I tried to implement a more flexible check system and I have some thoughts on the current structure of the dirty config setup:

We are currently going in a direction were we would need to spend a lot of time in maintaining everything that is related to `config.yaml`. This is because we are trying to cover every aspect individually (essentially the `missing_keys`). However if we add more options to `config.yaml` this will become a huge hassle, because we will need to add them to multiple functions and add a lot of `if-elif` statements. On the upside, this is the most convenient for the user, at the cost of being inconvenient for the developers.

The alternative would be to check if even a single options is missing and if it is, then we just prompt the user to re-create the `config.yaml`. This is much easier to expand and to maintain, but less convenient for the user. However I don't think that there will be many instances were the user will need to re-configure his `config.yaml` file.

So to summarize in which direction I would move:

1. Inside `check_config_data` check if even a single option is missing
2. If it is, then prompt user to re-configure his `config.yaml` (re-create it as if there was none to begin with)

We can even use a better way of checking all the options (without hard-coding them) separating a bit of code from `create_config_yaml()`.

**Comment by efJerryYang** [comment]
I have some ideas on how to improve the handling of `config.yaml` file. One approach is to maintain a global key list (or dict, because some of them are optional and we need values to specify those conditions, and it is also acceptable to use a global variable here for the entire package in my eyes) that should be covered in the configuration file, and thus an examination of missing keys can be performed over the keys we have in the global list or dict.

For the handling of missing keys, I consider that we can have a general function `setup_key(key: str, optional: bool) -> str:` to prompt the user to setup the missing keys, and we thus only this function needs to maintain a series of `if-elif` statements. Do you think this solution makes sense? I would be glad to explain, or you can leave this part to me (but I may need several days to update the code) and fix the panel part first.

**Comment by PraxTube** [comment]
My main concern is that we invest too much time into this rather small issue. These changes will only apply to a rather small amount of users (most likely the developers) and even then it wouldn't be super inconvenient to simple re-create the config instead of adding every single option manually. That said, if you want to have this functionality and are willing to invest the time into, I have no objections.

Both of the mentioned solutions seem fine in my eyes, they just require different levels of effort to implement and maintain.

**Comment by efJerryYang** [comment]
Well, I think we could just leave the problem here at present... :cry:

**Comment by PraxTube** [comment]
Yes agreed. This isn't very urgent.
